editor/selectors.js

```diff
@@ -8,48 +8,121 @@ import { first, last, get } from 'lodash';
  */
 import { addQueryArgs } from './utils/url';
 
+/**
+ * Returns the current editing mode.
+ *
+ * @param  {Object} state Global application state
+ * @return {String}       Editing mode
+ */
 export function getEditorMode( state ) {
 	return state.mode;
 }
 
+/**
+ * Returns true if the editor sidebar panel is open, or false otherwise.
+ *
+ * @param  {Object}  state Global application state
+ * @return {Boolean}       Whether sidebar is open
+ */
 export function isEditorSidebarOpened( state ) {
 	return state.isSidebarOpened;
 }
 
+/**
+ * Returns true if the past editor history snapshots exist, or false otherwise.
+ *
+ * @param  {Object}  state Global application state
+ * @return {Boolean}       Whether undo history exists
+ */
 export function hasEditorUndo( state ) {
 	return state.editor.history.past.length > 0;
 }
 
+/**
+ * Returns true if the future editor history snapshots exist, or false
+ * otherwise.
+ *
+ * @param  {Object}  state Global application state
+ * @return {Boolean}       Whether redo history exists
+ */
 export function hasEditorRedo( state ) {
 	return state.editor.history.future.length > 0;
 }
 
+/**
+ * Returns true if the currently edited post is yet to be saved, or false if
+ * the post has been saved.
+ *
+ * @param  {Object}  state Global application state
+ * @return {Boolean}       Whether the post is new
+ */
 export function isEditedPostNew( state ) {
 	return ! state.currentPost.id;
 }
 
+/**
+ * Returns true if there are unsaved values for the current edit session, or
+ * false if the editing state matches the saved or new post.
+ *
+ * @param  {Object}  state Global application state
+ * @return {Boolean}       Whether unsaved values exist
+ */
 export function isEditedPostDirty( state ) {
 	return state.editor.dirty;
 }
 
+/**
+ * Returns the post currently being edited.
```

Comment:
>Probably worth noting that this does not include any edits made to the post. "Returns our representation of the post in its last known state state on the server" maybe?

Raises an interesting point about what we'd expect the return value to be for a new post. Based on the [`currentPost` default](https://github.com/WordPress/gutenberg/blob/1e464894e30afae275895a1c40d4c124da3f30fd/editor/state.js#L184) I'd guess an empty object? Seems like it might be more appropriate as `null`, though I suspect this could cause breakage where we expect to access properties unguarded on the post object.

Nice thing about adding these documentation is that it helps surface real-world expectations 😄